src/main/scala/scalismo/common/Scalar.scala

```diff
@@ -408,4 +409,22 @@ object ScalarArray {
     }
   }
 
+  class ScalarVectorizer[S: Scalar] extends Vectorizer[S] {
+    override def dim: Int = 1
+
+    def toArray(v: S): Array[Double] = Array[Double](Scalar[S].toDouble(v))
+    override def vectorize(v: S): DenseVector[Double] = new DenseVector(toArray(v))
+
+    override def unvectorize(d: DenseVector[Double]): S = {
+      Scalar[S].fromDouble(d(0))
+    }
+
+    override def equals(that: Any): Boolean = {
```

Comment:
whenever equals is implemented, hashcode should also be implemented. 
But easier still would be to make it a case class

Reply:
I did not think about the possibility of making it a case class. My implementation is consistent with `scalismo.geometry.EuclideanVector.VectorVectorizer`.

Reply:
I made it a case class now. We should not forget to apply the same change to `scalismo.geometry.EuclideanVector.VectorVectorizer`.